uk.co.compucorp.civicrm.tasksassignments/js/test/services/task_test.js

```diff
@@ -0,0 +1,222 @@
+      });
+
+      it('returns saved Task on saving task', function () {
+        TaskService.save(fakeTask).then(function (data) {
+          expect(data).toEqual(fakeTask);
+        });
+
+        $httpBackend.flush();
+      })
+    });
+
+    describe("assign()", function () {
+      beforeEach(function () {
+        $httpBackend.whenGET(/action=getoptions&debug=true&entity=Task/).respond({});
+        $httpBackend.whenPOST(/action=copy_to_assignment&debug=true&entity=Task/).respond(taskMock.onAssign);
+      });
+
+      it('returns list of assigned tasks on task assign', function () {
+        TaskService.assign(fakeTaskList, 1).then(function (data) {
+          expect(data).toEqual(fakeTaskList);
+        });
+
+        $httpBackend.flush();
+      });
+    });
+
+    describe("saveMultiple()", function () {
+      beforeEach(function () {
+        $httpBackend.whenGET(/action=getoptions&debug=true&entity=Task/).respond({});
+        $httpBackend.whenPOST(/action=create_multiple&debug=true&entity=Task/).respond(taskMock.onSaveMultiple);
+      });
+
+      it('returns saved task list on saving multiple tasks', function () {
+        TaskService.saveMultiple(fakeTaskList).then(function (data) {
+          expect(data).toEqual(fakeTaskList);
+        });
+
+        $httpBackend.flush();
+      });
+    });
+
+    describe("sendReminder()", function () {
+      beforeEach(function () {
+        $httpBackend.whenGET(/action=getoptions&debug=true&entity=Task/).respond({});
+        $httpBackend.whenPOST(/action=sendreminder&debug=true&entity=Task/).respond(taskMock.onSendReminder);
+      });
+
+      it('returns reminder sent status on sending reminder for a task', function () {
+        TaskService.sendReminder(1, fakeReminderNote).then(function (data) {
+          expect(data.values).toEqual(true);
+        });
+
+        $httpBackend.flush();
+      });
+    });
+
+    describe("getActivityTypes()", function () {
+      beforeEach(function () {
+        $httpBackend.whenGET(/action=getoptions&debug=true&entity=Task/).respond(taskMock.onGetOptions.activityTypes);
+      });
+
+      beforeEach(function () {
+        promise = $q.all({
+          taskType: TaskService.getActivityTypes()
+        });
+      });
+
+      afterEach(function () {
+        $httpBackend.flush();
+      });
+
+      it("returns promise with array of list of activity types", function () {
+        promise.then(function (options) {
+          expect(options.taskType.obj).toBeDefined();
+          expect(options.taskType.arr).toEqual(fakeActivityTypes);
+        });
+      });
+    });
+
+    describe("getTaskStatus()", function () {
+      beforeEach(function () {
+        $httpBackend.whenGET(/action=getoptions&debug=true&entity=Task/).respond(taskMock.onGetOptions.taskStatus);
+      });
+
+      beforeEach(function () {
+        promise = $q.all({
+          taskStatus: TaskService.getTaskStatus()
+        });
+      });
+
+      afterEach(function () {
```

Comment:
Please check if `$httpBackend.flush();` is called for every test case, if yes then place it in the outer describe's afterEach and remove individual one's